hbase_sql/src/test/java/com/splicemachine/hbase/CostEstimationIT.java

```diff
@@ -460,4 +461,44 @@ public void testPreferIndexAccessEqualCosts() throws Exception {
 
         rowContainsQuery(new int[]{4, 5}, sqlText, methodWatcher, "IndexLookup", "IndexScan[EMPTY_IDX2");
     }
+
+    // DB-11860
+    @Test
+    public void testPreferIndexOnSpecialColumns() throws Exception {
+        methodWatcher.execute("CREATE TABLE \"TP\" (\n" +
+                "\"BP#\" CHAR(36) NOT NULL\n" +
+                ",\"GF#\" CHAR(36) NOT NULL\n" +
+                ",\"TRS\" TIMESTAMP NOT NULL\n" +
+                ",\"GFAK\" CHAR(5) NOT NULL DEFAULT ''\n" +
+                ",\"A_G#\" CHAR(36) NOT NULL\n" +
+                ",\"SN\" CHAR(18) NOT NULL DEFAULT ''\n" +
+                ",\"K_A\" CHAR(1) NOT NULL DEFAULT ''\n" +
+                ",\"ST5\" DECIMAL(5,0) NOT NULL DEFAULT 0.0\n" +
+                ",\"ST\" CHAR(1) NOT NULL DEFAULT ''\n" +
+                ",\"VAK\" CHAR(1) NOT NULL DEFAULT ''\n" +
+                ",\"FK#\" CHAR(36) NOT NULL\n" +
+                ",\"D\" VARCHAR (32000) FOR BIT DATA NOT NULL DEFAULT X''\n" +
+                ", CONSTRAINT PK PRIMARY KEY(\"BP#\"))");
+
+        methodWatcher.execute("create index IDX1 on TP(GF#, TRS, ST, VAK, GFAK, SN, A_G#)");
+        methodWatcher.execute("create index IDX2 on TP(SN, VAK, GF#, ST, TRS, GFAK, A_G#)");
```

Comment:
@ascend1 how does this work? Does IDX1 have 1 specialColumn (because of the predicate on GF#) and IDX2 0 because there's no prefix up to GF#? If IDX2 was (SN, GF#,...) would they be tied or would IDX1 win because it's first?

Reply:
The code counts how many scan start/stop keys are on special columns. If IDX2 is defined as (SN, GF#) then they will tie, because the predicate on GF# would also be a scan key. Order of column is not considered because generally, there is no relation between how selective a predicate is and on which index column ordinal position the predicate applies.

Reply:
That wasn't the relation I was going for, if you have two predicates P0 on C0 and P1 on C1, and you have idx0 (C0, C1) and idx1 (C1, C0), you want to choose the index that puts the most selective predicate first (that means you end up scanning less rows). That's what I meant.

Reply:
I see. Then I think our selectivity estimate to apply for calculating the number of scanned rows is not really correct. As far as I understand, if we have P0 on C0 and P1 on C1, both are scan start/stop keys, then they are pushed to the same scan selectivity phase. That means scan selectivity is s(P0) * sqrt(s(P1))  (suppose s(P0) is smaller), and it's the same for both idx0 and idx1. According to what you just mentioned, we should probably calculate this only based on predicates applied on the leading column?

Back to this ticket, unfortunately the selectivitiy of predicates are not available when we compare costs. They are calculated in `ScanCostFunction` and then thrown away. But again, this is not a proper fix anyway and we need a special treatment for the 0-stat case.